packages/ws-server/src/webSocketServer.ts

```diff
@@ -46,11 +46,20 @@ const LOGGER_PREFIX = 'WebSocket:';
 
 const CHAIN_ID = relay.eth().chainId();
 const DEFAULT_ERROR = predefined.INTERNAL_ERROR();
+const MAX_CONNECTIONS = parseInt(process.env.CONNECTION_LIMIT || '10');
+let connectedClients = 0;
 
 app.ws.use((ctx) => {
     ctx.websocket.id = relay.subs()?.generateId();
     logger.info(`${LOGGER_PREFIX} new connection ${ctx.websocket.id}`);
 
+    if (connectedClients >= MAX_CONNECTIONS) {
+        ctx.websocket.send("Connections exceeded");
+        ctx.websocket.close();
+        return;
+    }
+    connectedClients++;
```

Comment:
Q: does the server or middleware itself not have a method to get the number of connected clients?
If it does we could just utilize that instead of our own counter which gets us into the counting problem noted below.

If not then maybe the counter should increment prior to the max connection check since it is alive.
In this way the close decrement logic will be consistent

Reply:
The koa-websockets package is a koa wrapper around the JS websockets (ws) package.  I am not seeing anything that tracks the number of connections in the WS package.  The only related identifier is the individual client IP address, which will not work behind a proxy.

Reply:
Sounds good then incrementing before  check should fix the issue

Reply:
Yes, it does, but it's not in the koa-websockets, it's in the node websockets package that the koa-websockets wraps.  The latest commit now uses the verifyClient attribute in the constructor to pass in a callback function that handles the connections.